netbox/dcim/tests/test_traces.py

```diff
@@ -0,0 +1,250 @@
+                device_role=self.device_role['patch'],
+            )
+            self.patch_rp[patch_nr] = {}
+            self.patch_fp[patch_nr] = {}
+            for port_nr in (1, 2, 3, 4):
+                self.patch_rp[patch_nr][port_nr] = RearPort.objects.create(
+                    device=self.patch[patch_nr],
+                    name="Panel{} RP{}".format(patch_nr, port_nr),
+                    type=PortTypeChoices.TYPE_SC,
+                    positions=1
+                )
+                self.patch_fp[patch_nr][port_nr] = FrontPort.objects.create(
+                    device=self.patch[patch_nr],
+                    name="Panel{} FP{}".format(patch_nr, port_nr),
+                    type=PortTypeChoices.TYPE_SC,
+                    rear_port=self.patch_rp[patch_nr][port_nr],
+                    rear_port_position=1
+                )
+
+        # DWDM MUXes with 8 channels each
+        self.mux = {}
+        self.mux_rp = {}
+        self.mux_fp = {}
+        for mux_nr in (1, 2, 3, 4):
+            self.mux[mux_nr] = Device.objects.create(
+                name="MUX{}".format(mux_nr),
+                site=self.site,
+                device_type=self.device_type['mux'],
+                device_role=self.device_role['mux'],
+            )
+
+            self.mux_rp[mux_nr] = RearPort.objects.create(
+                device=self.mux[mux_nr],
+                name="MUX{} DWDM".format(mux_nr),
+                type=PortTypeChoices.TYPE_LC,
+                positions=8
+            )
+
+            self.mux_fp[mux_nr] = {}
+            for port_nr in (1, 2, 3, 4, 5, 6, 7, 8):
+                self.mux_fp[mux_nr][port_nr] = FrontPort.objects.create(
+                    device=self.mux[mux_nr],
+                    name="MUX{} CH{}".format(mux_nr, port_nr),
+                    type=PortTypeChoices.TYPE_LC,
+                    rear_port=self.mux_rp[mux_nr],
+                    rear_port_position=port_nr
+                )
+
+        # And a circuit
+        self.provider = Provider.objects.create(
+            name='Provider 1',
+            slug='provider-1',
+        )
+        self.circuit_type = CircuitType.objects.create(
+            name='Circuit Type 1',
+            slug='circuit-type-1'
+        )
+        self.circuit = Circuit.objects.create(
+            cid='Circuit 1',
+            provider=self.provider,
+            type=self.circuit_type
+        )
+        self.circuit_term_a = CircuitTermination.objects.create(
+            circuit=self.circuit,
+            port_speed=1,
+            term_side=CircuitTerminationSideChoices.SIDE_A,
+            site=self.site
+        )
+        self.circuit_term_z = CircuitTermination.objects.create(
+            circuit=self.circuit,
+            port_speed=1,
+            term_side=CircuitTerminationSideChoices.SIDE_Z,
+            site=self.site
+        )
+
+        # And connect everything together in a way that covers all combinations:
```

Comment:
Can you add an ASCII topology diagram here for clarity? http://asciiflow.com/ works great for this. Or if you can sketch something out in Visio or similar I'd be happy to convert it.

Reply:
Included in the next commit